Src/Library/Main/MainExtensions.cs

```diff
@@ -60,6 +63,9 @@ public static IApplicationBuilder UseFastEndpoints(this IApplicationBuilder app,
 
     public static IEndpointRouteBuilder MapFastEndpoints(this IEndpointRouteBuilder app, Action<Config>? configAction = null)
     {
+        //use AntiforgeryMiddleware middleware
+        (app as WebApplication)?.UseMiddleware<Middleware.AntiforgeryMiddleware>();
```

Comment:
Maybe only use middleware if antiforgery is used?

Could be done by checking if its turned "on" on any endpoint, or by some other configuration

Reply:
> Maybe only use middleware if antiforgery is used?
> 
> Could be done by checking if its turned "on" on any endpoint, or by some other configuration

the Middlewire is really Simple, just when set EndpointDefinition.IsEnlableAntiforgery = true  the validation logic is executed

 (Default:false)  it has no effect  :)

```c#
    public async Task Invoke(HttpContext context)
    {
        //GET请求不需要防伪验证
        if (context.Request.Method == HttpMethods.Get ||
            context.Request.Method == HttpMethods.Trace ||
            context.Request.Method == HttpMethods.Options ||
            context.Request.Method == HttpMethods.Head)
        {
            await _next(context);
            return;
        }
        var endpointDefinition = context.GetEndpoint()?.Metadata.GetMetadata<EndpointDefinition>();
        if (endpointDefinition?.IsEnlableAntiforgery is true)
        {
            try
            {
                await _antiforgery.ValidateRequestAsync(context);
            }
            catch (AntiforgeryValidationException)
            {
                context.Response.StatusCode = StatusCodes.Status400BadRequest;
                await context.Response.WriteAsync("Invalid anti-forgery token");
                return;
            }
        }
        await _next(context);
    }
```

Reply:
haven't checked the code properly yet. but ideally this middleware should only be added to the pipeline if user explicitly asks to enable it during startup, something like this:
```cs
app.UseFastEndpoints(c => c.Security.EnableAntiForgeryTokens = true);
```
if this value is `false`, endpoint level `EnlableAntiforgery()` call should throw an exception during startup.

@vipwan think you can make that happen?

Reply:
actually scratch that.... lets make the user do two calls to configure antiforgery.

```cs
builder.Services.AddAntiForgery()

app.UseAntiForgery() // this will be a custom extensionmethod that registers the middleware.
```

and during startup if endpoints say `EnableAntiForgery()` and the user hasn't called `AddAntiForgery()` an exception should be thrown.

how about that?

Reply:
> haven't checked the code properly yet. but ideally this middleware should only be added to the pipeline if user explicitly asks to enable it during startup, something like this:
> 
> ```cs
> app.UseFastEndpoints(c => c.Security.EnableAntiForgeryTokens = true);
> ```
> 
> if this value is `false`, endpoint level `EnlableAntiforgery()` call should throw an exception during startup.
> 
> @vipwan think you can make that happen?

its a good Idea
```c#
app.UseFastEndpoints(c => c.Security.EnableAntiForgeryTokens = true);
```

Reply:
i think this should be the way:
```cs
builder.Services.AddAntiForgery()

app.UseAntiForgery() // this will be a custom extensionmethod that registers the middleware.
```